core/src/main/java/org/htsjdk/core/api/PathURI.java

```diff
@@ -0,0 +1,86 @@
+package org.htsjdk.core.api;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.file.Path;
+
+/**
+ * Interface representing htsjdk-next input/output resources as URIs.
+ */
+public interface PathURI {
+
+    /**
+     * @return true if this URI has a scheme that has an installed NIO file system provider. This does not
+     * guarantee the URI can be converted into a {@code java.nio.file.Path}, since the URI can be syntactically
+     * valid, and specify a valid file system provider, but still fail to be semantically meaningful.
+     */
+    boolean isNIO();
+
+    /**
+     * Return true if this {code PathURI} can be resolved to an NIO Path. If true, {@code #toPath()} can be
+     * safely called.
+     *
+     * There are cases where a valid URI with a valid scheme backed by an installed NIO File System
+     * still can't be turned into a {@code java.nio.file.Path}, i.e., the following specifies an invalid
+     * authority "namenode":
+     *
+     *  hdfs://namenode/to/file
+     *
+     * The current implementation returns false for these cases (toPath will fail, getInvalidPathReason
+     * returns the reason code).
+     */
+    boolean isPath();
```

Comment:
Do we really need `isNio` and `isPath`? I know that they are slightly different (if they are valid or not), but I prefer to just throw detailed exceptions if a `Path` is retrieved.

Reply:
Callers can still fo the path conversion themselves if desired.